sphinxcontrib/blockdiag.py

```diff
@@ -219,20 +219,41 @@ def html_depart_blockdiag(self, node):
 
 
 def get_image_format_for(builder):
-    if builder.format in ('html', 'slides'):
-        image_format = builder.config.blockdiag_html_image_format.upper()
-    elif builder.format == 'latex':
-        if builder.config.blockdiag_tex_image_format:
-            image_format = builder.config.blockdiag_tex_image_format.upper()
-        else:
-            image_format = builder.config.blockdiag_latex_image_format.upper()
+    config = builder.config
+    image_formats = config.blockdiag_image_format or {}
+
+    # Find image format in a configured dictionary; or assume that the
+    # provided configuration is a string format (for convenience).
```

Comment:
I feel such fallback is not needed. Let it crash.

Reply:
Will do ~ the logic has been simplified to expect the dictionary value and removed the str-only cases.